backend/lucene/src/main/java/org/hibernate/search/backend/lucene/document/impl/LuceneNestedObjectDocumentBuilder.java

```diff
@@ -28,6 +52,7 @@ void contribute(String rootIndexName, MultiTenancyStrategy multiTenancyStrategy,
 		document.add( new StringField( LuceneFields.typeFieldName(), LuceneFields.TYPE_CHILD_DOCUMENT, Store.YES ) );
 		document.add( new StringField( LuceneFields.rootIndexFieldName(), rootIndexName, Store.YES ) );
 		document.add( new StringField( LuceneFields.rootIdFieldName(), rootId, Store.YES ) );
+		document.add( new Field( LuceneFields.rootIdBinaryFieldName(), new BytesRef( rootId ), TYPE_STORED_BINARY ) );
```

Comment:
You don't need to add a new field, you can just replace the one above with this one. Here you're storing two values in each document for the root ID field name, which is not useful.
Just replace this:

```
		document.add( new StringField( LuceneFields.rootIdFieldName(), rootId, Store.YES ) );
		document.add( new Field( LuceneFields.rootIdBinaryFieldName(), new BytesRef( rootId ), TYPE_STORED_BINARY ) ); 
```

... with this:

```
		document.add( new Field( LuceneFields.rootIdFieldName(), new BytesRef( rootId ), TYPE_STORED_BINARY ) ); 
```

And remove the method `LuceneFields.rootIdBinaryFieldName`.

Reply:
I had figured out that it could have been replaced. But I wasn't sure about.